src/modules/peek/Peek.FilePreviewer/FilePreview.xaml.cs

```diff
@@ -93,7 +94,11 @@ private async Task OnFilePropertyChanged()
                 return;
             }
 
-            Previewer = previewerFactory.Create(File);
+            // Cancel previous loading task
+            _cancellationTokenSource.Cancel();
+            _cancellationTokenSource = new ();
+
+            Previewer = previewerFactory.Create(File, _cancellationTokenSource.Token);
```

Comment:
I think the token should be passed in the async methods. This is the established pattern in C# and would also clean up the dependency of the preview factory and the previewers!